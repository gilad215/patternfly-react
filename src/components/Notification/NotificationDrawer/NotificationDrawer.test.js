import React from 'react';
import renderer from 'react-test-renderer';
import { NotificationDrawer, NotificationDrawerToggle } from './index';

test('NotificationDrawer is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer>
      <div>Child</div>
    </NotificationDrawer>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Title is working properly', () => {
  const component = renderer.create(<NotificationDrawer.Title />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Accordion is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.Accordion>
      <div>Child</div>
    </NotificationDrawer.Accordion>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Toggle is working properly', () => {
  const component = renderer.create(<NotificationDrawerToggle />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Dropdown is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.Dropdown id="1">
      <div>Child</div>
    </NotificationDrawer.Dropdown>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.Panel>
      <div>Child</div>
    </NotificationDrawer.Panel>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel Action and Link is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.PanelAction>
      <NotificationDrawer.PanelActionLink>
        <div>Child</div>
      </NotificationDrawer.PanelActionLink>
    </NotificationDrawer.PanelAction>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel Body is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.PanelBody>
      <div>Child</div>
    </NotificationDrawer.PanelBody>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel Counter is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.PanelCounter>
      <div>Child</div>
    </NotificationDrawer.PanelCounter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel Heading is working properly', () => {
  const component = renderer.create(
    <NotificationDrawer.PanelHeading>
      <div>Child</div>
    </NotificationDrawer.PanelHeading>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NotificationDrawer Panel Title is working properly', () => {
  const component = renderer.create(<NotificationDrawer.PanelTitle />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
