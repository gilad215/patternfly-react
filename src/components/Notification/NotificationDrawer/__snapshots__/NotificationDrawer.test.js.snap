// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawer Accordion is working properly 1`] = `
<div
  className="panel-group"
  id="notification-drawer-accordion"
>
  <div>
    Child
  </div>
</div>
`;

exports[`NotificationDrawer Dropdown is working properly 1`] = `
<div
  className="dropdown pull-right dropdown-kebab-pf dropdown btn-group"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-link"
    disabled={false}
    id="1"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    type="button"
  >
    <span
      aria-hidden={true}
      className="fa fa-ellipsis-v"
    />
  </button>
  <ul
    aria-labelledby="1"
    className="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <div
      onKeyDown={[Function]}
      onSelect={[Function]}
    >
      Child
    </div>
  </ul>
</div>
`;

exports[`NotificationDrawer Panel Action and Link is working properly 1`] = `
<div
  className="drawer-pf-action"
>
  <div
    className="drawer-pf-action-link"
  >
    <div>
      Child
    </div>
  </div>
</div>
`;

exports[`NotificationDrawer Panel Body is working properly 1`] = `
<div
  className="panel-body"
>
  <div>
    Child
  </div>
</div>
`;

exports[`NotificationDrawer Panel Counter is working properly 1`] = `
<span
  className="panel-counter"
>
  <div>
    Child
  </div>
</span>
`;

exports[`NotificationDrawer Panel Heading is working properly 1`] = `
<div
  className="panel-heading"
>
  <div>
    Child
  </div>
</div>
`;

exports[`NotificationDrawer Panel Title is working properly 1`] = `
<h4
  className="panel-title"
/>
`;

exports[`NotificationDrawer Panel is working properly 1`] = `
<div
  className="panel panel-default"
>
  <div>
    Child
  </div>
</div>
`;

exports[`NotificationDrawer Title is working properly 1`] = `
<div
  className="drawer-pf-title"
>
  <a
    className="drawer-pf-toggle-expand"
    onClick={null}
  />
  <a
    className="drawer-pf-close"
    onClick={null}
  >
    <span
      aria-hidden={true}
      className="fa fa-close"
    />
  </a>
  <h3
    className="text-center"
  >
    Notifications
  </h3>
</div>
`;

exports[`NotificationDrawer Toggle is working properly 1`] = `
<button
  className="btn btn-link"
  disabled={false}
  onBlur={[Function]}
  onClick={null}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <span
    aria-describedby="tooltip"
    aria-hidden={true}
    className="fa fa-bell fa-lg"
    style={
      Object {
        "color": "rgb(240, 240, 240)",
        "marginTop": "20px",
      }
    }
  />
</button>
`;

exports[`NotificationDrawer is working properly 1`] = `
<div
  className="drawer-pf drawer-pf-notifications-non-clickable"
>
  <div>
    Child
  </div>
</div>
`;
